impl/client/src/main/javascript/web/pentaho/action/Base.js

```diff
@@ -80,51 +80,96 @@ define([
      * Gets the type of action.
      *
      * @name type
-     * @memberOf pentaho.action.Abstract#
+     * @memberOf pentaho.action.Base#
      * @type {string}
+     * @readonly
      *
      * @abstract
      */
     get type() {
     },
 
+    /**
+     * Gets the event name of the action.
+     *
+     * @name eventName
+     * @memberOf pentaho.action.Base#
+     * @type {string}
+     * @readonly
+     *
+     * @abstract
+     */
     get eventName() {
-      return this.type;
     },
 
     /**
+     * Determines if the given action is valid.
+     *
+     * The default implementation does nothing and considers the instance valid.
+     * Override to implement an action's specific validation logic.
+     *
      * @name validate
-     * @memberOf pentaho.action.Abstract#
+     * @memberOf pentaho.action.Base#
      *
      * @abstract
+     * @method
+     * @return {Array.<pentaho.lang.UserError>} A non-empty array of errors or `null`.
      */
     validate: function() {
     },
 
     /**
+     * Creates a shallow clone of this value.
+     *
      * @name clone
-     * @memberOf pentaho.action.Abstract#
+     * @memberOf pentaho.action.Base#
      *
      * @abstract
+     * @method
+     * @return {pentaho.action.Base} The action clone.
      */
     clone: function() {
+      return new this.constructor(this.toSpec());
     },
 
     /**
+     * Creates a specification that describes this action.
+     *
      * @name toSpec
-     * @memberOf pentaho.action.Abstract#
+     * @memberOf pentaho.action.Base#
      *
-     * @abstract
+     * @method
+     * @return {pentaho.action.spec.IBase} An action specification.
      */
     toSpec: function() {
+      var spec = {};
+
+      this._fillSpec(spec);
+
+      return spec;
+    },
+
+    /**
+     * Fills the given specification with this action's attributes' local values,
+     * and returns whether any attribute was actually added.
+     *
+     * @param {pentaho.action.spec.IBase} [spec] An action specification.
+     * @protected
+     * @abstract
+     */
+    _fillSpec: function(spec) {
```

Comment:
Better use the same convention as other doclets and just remove the implementation, adding `@name` and `@memberOf`.